### PR TITLE
disconnecting connection

### PR DESCRIPTION
**Pull request by BenediktNiehues**
according to http://docs.oracle.com/javase/8/docs/technotes/guides/net/http-keepalive.html avoid abandoning the HttpURLConnection by reading the errorStream and calling disconnect on the connection to close the undrlying socket if its not persistent.

### CODE REVIEW

File: bundles/org.jupnp/src/main/java/org/jupnp/http/HttpFetch.java [hunk 1]
```diff
@@ -73,12 +75,23 @@
...
                 //String responseBody = IO.readString(urlConnection.getErrorStream());
             }
             throw ex;
         } finally {
-            if (is != null) is.close();
+            if (is != null) {
+                is.close();
+            }
+            if (urlConnection != null) {
+                urlConnection.disconnect();
```
Comment (afuechsel) [inline bundles/org.jupnp/src/main/java/org/jupnp/http/HttpFetch.java]: If this method is called very often, the connection could not be reused.

**Comment by BenediktNiehues** [comment]
@afuechsel you are right, this class is not used, at least in our context. So we should trigger a discussion to remove this unused code.

File: bundles/org.jupnp/src/main/java/org/jupnp/http/HttpFetch.java (hunk 1 above)
Comment (BenediktNiehues) [inline bundles/org.jupnp/src/main/java/org/jupnp/http/HttpFetch.java]: the underlying socket is only closed if the connection is not persistent...

**Comment by kaikreuzer** [comment]
> So we should trigger a discussion to remove this unused code.

It is in an exported package and therefore part of the public API. According to semver, we should therefore only remove it once we go to version 3.0 of jUPnP.

**Pull request merged**